variables.tf

```diff
@@ -48,22 +57,22 @@ variable "region" {
   description = "The region this bucket should reside in, defaults to the region used by the callee"
 }
 
-variable "cors_rule" {
-  type = object({
-    allowed_headers = list(string)
-    allowed_methods = list(string)
-    allowed_origins = list(string)
-    expose_headers  = list(string)
-    max_age_seconds = number
-  })
-  default     = null
-  description = "The CORS rule for the s3 bucket"
+variable "replication_configuration" {
+  type        = any
+  default     = {}
+  description = "Map containing cross-region replication configuration."
```

Comment:
Can we make this a specific object (just like is done with the `cors_rule` variable)? This would be a super nice improvement in understanding the available options...